pkg/pdutil/pd.go

```diff
@@ -143,21 +146,30 @@ func pdRequest(
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	resp, err := cli.Do(req)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	defer resp.Body.Close()
-	if resp.StatusCode != http.StatusOK {
-		res, _ := ioutil.ReadAll(resp.Body)
-		return nil, errors.Annotatef(berrors.ErrPDInvalidResponse, "[%d] %s %s", resp.StatusCode, res, reqURL)
-	}
+	var (
+		resp   *http.Response
+		reqErr error
+	)
 
-	r, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return nil, errors.Trace(err)
+	for i := 0; i < pdRequestRetryTime; i++ {
+		resp, reqErr = cli.Do(req)
+		if err != nil {
+			log.Warn("pd request fail, retry", zap.Int("retry time", i), zap.Error(reqErr))
+			time.Sleep(time.Duration(i) * time.Second)
```

Comment:
I think just sleep 1 second is better. then if there is something wrong with PD. we can wait at most 10 seconds.